packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx

```diff
@@ -130,15 +134,26 @@ export class ReactAvailableViewsWidget extends React.Component<ReactAvailableVie
     private async fetchAnnotationCategories(output: OutputDescriptor) {
         const tspClient = this.props.tspClientProvider.getTspClient();
         if (this._selectedExperiment) {
-            const annotationCategories = await tspClient.fetchAnnotationsCategories(this._selectedExperiment?.UUID, output.id);
+            const annotationCategories = await tspClient.fetchAnnotationsCategories(this._selectedExperiment.UUID, output.id, this.markerSetId);
             const annotationCategoriesResponse = annotationCategories.getModel();
             if (annotationCategories.isOk() && annotationCategoriesResponse) {
                 const categoriesList = annotationCategoriesResponse.model ? annotationCategoriesResponse.model.annotationCategories : [];
-                signalManager().fireAnnotationsFetchedSignal(categoriesList);
+                const payload = {
+                    traceViewerId: this._selectedExperiment.UUID,
+                    outputId: output.id,
+                    annotationCategories: categoriesList
```

Comment:
If you have an output opened and disable a specific category, then you open a new output that has the same category, this code will 'turn on' the category that the user had hidden, but the first output doesn't update its annotations (note: it shouldn't have to). Then you have two outputs with out-of-sync annotation categories, although they share the same Markers filter toolbar button.